oak_runtime/src/introspect.rs

```diff
@@ -25,6 +25,71 @@ use prost::Message;
 use regex::Regex;
 use std::{net::SocketAddr, sync::Arc};
 
+#[cfg(not(feature = "oak_introspection_client"))]
+mod introspection_client {
+    use super::*;
+
+    pub fn find_client_file(_path: &str) -> Option<Response<Body>> {
+        None
+    }
+}
+
+#[cfg(feature = "oak_introspection_client")]
+mod introspection_client {
+    use super::*;
+    use hyper::header::CONTENT_ENCODING;
+
+    fn static_file(
+        file: &str,
+        content_type: &str,
+        content_encoding: Option<&str>,
+    ) -> Response<Body> {
+        let mut response = Response::new(Body::from(file.to_string()));
+        let headers = response.headers_mut();
+        headers.insert(CONTENT_TYPE, content_type.parse().unwrap());
+        if let Some(content_encoding_value) = content_encoding {
+            headers.insert(CONTENT_ENCODING, content_encoding_value.parse().unwrap());
```

Comment:
Is it enough to just add this header in order to make the server actually do the compression magically? Or is there something else that I am missing? Either way, perhaps worth a comment / link to some reference?

Reply:
Good point, that's not really obvious from the code, happy to add a comment. 

The tl;dr is that the rust server only serves static files, headers can be seen as mere "metadata" instructing the browser how to parse the received files. 

The gzip compression itself takes place when assets are generated by webpack.

Reply:
Ah thanks, my bad, I didn't notice the `.gz` suffix earlier, it all makes sense now.